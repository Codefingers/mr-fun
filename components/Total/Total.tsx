import React from 'react';

import styled from 'styled-components';
import { Product } from '../../types';

const calculateTotal = (basket) =>
  basket?.length &&
  basket.reduce((total, item) => {
    return total + item.price * item.quantity;
  }, 0);

const TotalContainer = styled.div`
  margin-top: 32px;
  padding: 0;
  border-top: 4px solid;
  display: flex;
  flex-direction: column;
  align-items: flex-end;

  h2 {
    display: block;
  }
`;

export const TotalComponent: React.FC<{ basket: Array<Product> }> = ({
  basket,
}) => (
  <TotalContainer>
    <h2>Total: {`Â£${calculateTotal(basket)}`}</h2>
  </TotalContainer>
);

export default TotalComponent;
