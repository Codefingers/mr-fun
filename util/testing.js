import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={{
        colors: {
          body: '#838fa6',
          heading: '#ebedf2',
          accent: '#BA99FF',
          button: '#FFF',
        },
      }}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
