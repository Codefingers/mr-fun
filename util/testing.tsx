import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const Providers = ({ children }) => {
  return (
    <ThemeProvider
      theme={theme}
    >
      {children}
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

// override render method
export { customRender as render };
