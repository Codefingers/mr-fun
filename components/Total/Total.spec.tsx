import React from 'react';
import { render } from '../../util/testing';

import Total from './Total';

describe('<Total />', () => {
  const props = {
    basket: [
      {
        id: '1',
        price: 10.0,
        quantity: 2,
        title: 'product 1',
      },
      {
        id: '2',
        price: 12.4,
        quantity: 1,
        title: 'product 2',
      },
    ],
  };

  it('renders with correct total', () => {
    const { container } = render(<Total {...props} />);
    expect(container).toMatchSnapshot();
  });
});
