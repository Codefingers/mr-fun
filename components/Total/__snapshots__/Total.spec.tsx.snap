// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Total /> renders with correct total 1`] = `
<div>
  <div
    class="Total__TotalContainer-sc-149mlyd-0 kkKrkF"
  >
    <h2>
      Total: 
      Â£32.4
    </h2>
  </div>
</div>
`;
